File: bo-bonusesrewards-microservice/src/main/java/world/inclub/bonusesrewards/shared/payment/application/dto/PaymentListView.java
package world.inclub.bonusesrewards.shared.payment.application.dto;

import lombok.Builder;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Builder
public record PaymentListView(
        UUID paymentId,
        String username,
        String memberFullName,
        String nrodocument,
        String operationNumber,
        Long bonusTypeId,
        String bonusTypeName,
        Integer installmentNum,
        BigDecimal subTotalAmount,
        BigDecimal commissionAmount,
        BigDecimal rateAmount,
        BigDecimal totalAmount,
        LocalDate dueDate,
        LocalDateTime paymentDate,
        String voucherImageUrl,
        Long paymentStatusId,
        String paymentStatusName
) {}

File: bo-admin/src/app/manage-prize/interface/payment-list-view.ts
export interface IPaymentListViewResponse {
  result: boolean;
  data: IPaymentListViewData;
  timestamp: string;
  status: number;
}

export interface IPaymentListViewData {
  content: IPaymentListView[];
  currentPage: number;
  totalPages: number;
  totalElements: number;
  pageSize: number;
  hasNext: boolean;
  hasPrevious: boolean;
  first: boolean;
  last: boolean;
  numberOfElements: number;
  sortBy: string;
  sortDirection: string;
}

export interface IPaymentListView {
  paymentId: string;
  username: string;
  memberFullName: string;
  nrodocument: string;
  operationNumber: string;
  bonusTypeId: number;
  bonusTypeName: string;
  installmentNum: number;
  subTotalAmount: number;
  commissionAmount: number;
  rateAmount: number;
  totalAmount: number;
  dueDate: string;
  paymentDate: string;
  voucherImageUrl: string;
  paymentStatusId: number;
  paymentStatusName: string;
}